File: frontend/src/types/loan.ts
export interface Loan {
  id: string;
  amount: number;
  termMonths: number;
  interestRate: number;
  purpose: string;
  status: 'pending' | 'active' | 'completed' | 'defaulted';
  borrowerId: string;
  lenderId?: string;
  riskScore: number;
  createdAt: string;
  currentBalance: number;
  repaymentSchedule: RepaymentSchedule[];
}

export interface RepaymentSchedule {
  dueDate: string;
  amount: number;
  status: 'pending' | 'paid' | 'overdue';
}

export interface LoanState {
  loans: Loan[];
  loading: boolean;
  error: string | null;
} 

File: backend/app/api/models/loan.py
from pydantic import BaseModel
from typing import Optional, List
from datetime import datetime

class LoanBase(BaseModel):
    amount: float
    term_months: int
    interest_rate: float
    purpose: str

class LoanCreate(LoanBase):
    borrower_id: str

class LoanInDB(LoanBase):
    id: str
    borrower_id: str
    lender_id: Optional[str]
    status: str
    risk_score: float
    created_at: datetime
    repayment_schedule: List[dict]
    current_balance: float 